lib/command-publish.js

```diff
@@ -42,29 +43,54 @@ let _hostingInfo
 let _testFlag
 let _cloudFrontFlag
 
-function publish(testFlag, cloudFrontFlag) {
+function run(testFlag, cloudFrontFlag) {
+    _testFlag = testFlag
+    _cloudFrontFlag = cloudFrontFlag
     _projectInfo = projectInfoManager.getProjectInfo()
     if(_projectInfo){
-        awsConfigManager.checkAWSConfig(function(awsDetails){
-            _testFlag = testFlag
-            _cloudFrontFlag = cloudFrontFlag
-            _awsDetails = awsDetails
-            _backendDetails = backendInfoManager.getBackendDetails(_projectInfo.ProjectPath)
-            if(_backendDetails){
-                _hostingInfo = hostingOps.getInfo(_projectInfo, _awsDetails.config, _backendDetails)
-                if(_hostingInfo){
-                    projectAppBuilder.build(function(){
-                        uploadFiles()
-                    })
-                }else{
-                    console.log(chalk.red('no hosting feature'))
-                    console.log('please enable hosting and try again')
+        if(suitableForPublish(_projectInfo)){
```

Comment:
Maybe you can add this on welcome message to not add that command if the is not suitable for publish